fix: filter invalid issues before other processors

ldez opened the pull request:
Before:

```console
$ golangci-lint run 
WARN [runner] Can't process result by autogenerated_exclude processor: can't filter issue result.Issue{FromLinter:"contextcheck", Text:"Function `NewWriteLock$1` should pass the context parameter", Severity:"", SourceLines:[]string(nil), Replacement:(*result.Replacement)(nil), Pkg:(*packages.Package)(0xc00012c480), LineRange:(*result.Range)(nil), Pos:token.Position{Filename:"", Offset:0, Line:0, Column:0}, HunkPos:0, ExpectNoLint:false, ExpectedNoLintLinter:""}: no file path for issue 
WARN [runner] Can't process result by nolint processor: can't filter issue result.Issue{FromLinter:"contextcheck", Text:"Function `NewWriteLock$1` should pass the context parameter", Severity:"", SourceLines:[]string(nil), Replacement:(*result.Replacement)(nil), Pkg:(*packages.Package)(0xc00012c480), LineRange:(*result.Range)(nil), Pos:token.Position{Filename:"", Offset:0, Line:0, Column:0}, HunkPos:0, ExpectNoLint:false, ExpectedNoLintLinter:""}: no file path for issue 
WARN [runner/source_code] Failed to get line 0 for file : failed to get file  lines cache: can't get file  bytes from cache: can't read file : open : no such file or directory 
:0: Function `NewWriteLock$1` should pass the context parameter (contextcheck)
```

With this PR:
```console
$ ./golangci-lint run
WARN [runner/invalid_issue] no file path for the issue: probably a bug inside the linter "contextcheck": &result.Issue{FromLinter:"contextcheck", Text:"Function `NewWriteLock$1` should pass the context parameter", Severity:"", SourceLines:[]string(nil), Replacement:(*result.Replacement)(nil), Pkg:(*packages.Package)(0xc000be4480), LineRange:(*result.Range)(nil), Pos:token.Position{Filename:"", Offset:0, Line:0, Column:0}, HunkPos:0, ExpectNoLint:false, ExpectedNoLintLinter:""}
```

The previous behavior was causing a side effect on the `Nolint` processor.

**EDIT:** the invalid issues reported by `contextcheck` will not produce a warning because it's a known bug. 

Fixes #4550

---

_Sponsoring is a good way to sustain open source maintainers: [sponsor me](https://github.com/sponsors/ldez)_

alexandear commented on pkg/result/processors/invalid_issue.go:
```diff
@@ -0,0 +1,54 @@
+	if issue.FilePath() == "" {
+		// contextcheck has a known bug https://github.com/kkHAIKE/contextcheck/issues/21
+		if issue.FromLinter != "contextcheck" {
+			p.log.Warnf("no file path for the issue: probably a bug inside the linter %q: %#v", issue.FromLinter, issue)
```
Perhaps we should show this message as Info, as there is nothing the user can do in this situation.

ldez replied:
They can report the issue.

ldez commented:
For the details:
Previously inside the `AutogeneratedExclude` processor, this `issue.FilePath() == ""` was evaluated after this `filepath.Ext(name) == ".go"` but it's impossible for a file path to be `xx.go` and empty at the same time.

When I added the new option for the `AutogeneratedExclude` processor, I fixed the behavior.

The problem is that the `AutogeneratedExclude` processor was excluding reports not related to generated files.

And so welcome to side effects.
